ocaml/xapi/repository.ml

```diff
@@ -149,26 +149,69 @@ let get_proxy_params ~__context repo_name =
   | _ ->
       ("", "", "")
 
-let sync ~__context ~self ~token ~token_id =
+let sync ~__context ~self ~token ~token_id ~username ~password =
   try
     let repo_name = get_remote_repository_name ~__context ~self in
     remove_repo_conf_file repo_name ;
-    let binary_url, source_url =
-      match Db.Repository.get_origin ~__context ~self with
+    let origin = Db.Repository.get_origin ~__context ~self in
+
+    let binary_url, source_url, repo_gpgcheck, client_auth, server_auth =
+      match origin with
       | `remote ->
           ( Db.Repository.get_binary_url ~__context ~self
           , Some (Db.Repository.get_source_url ~__context ~self)
+          , true
+          , (CdnTokenAuth {token_id; token}, "accesstoken")
```

Comment:
If both the `token_id` and `token` are empty string, it could return `NoAuth` here. This avoids the empty `token_path` and checking against it later.
And the inconsistency like non-empty `token_id` and empty `token` will result in other error (like auth failure) raised from the CDN server side. This is expected. So we could rely on that instead of checking it in xapi. This will make the `with_sync_client_auth` and generating plugin parameter simpler as it will not be necessary to checking something like:
```
match token_path with
| "" ->
    ""
```
and 
```
  match auth with
  | CdnTokenAuth {token_id; token}, plugin when token_id = "" && token = "" ->
      f (Some ("", plugin))
  | CdnTokenAuth {token_id; token}, plugin when token_id <> "" && token <> "" ->
```